src/main/java/com/tterrag/k9/commands/CommandCurseGradle.java

```diff
@@ -0,0 +1,309 @@
+	 * Used to get the latest file from a curseforge project, of a particular minecraft version, then add it to the {@code list}
+	 * @param projectURL The projects url page. This should be the homepage, for example {@link https://minecraft.curseforge.com/projects/secretroomsmod}
+	 * @param MCVersion The minecraft version to use to get the latest version
+	 * @param waitMsg The message to use for infomation
+	 * @param ctx The Command Context
+	 * @param list A list to add the result to
+	 * @param page The files page to track. If you're calling this, the page should be 0 or 1. 
+	 * @throws CommandException 
+	 */
+	private void addLatestToList(String projectURL, String MCVersion, IMessage waitMsg, CommandContext ctx, ArrayList<CurseResult> list, int page) throws CommandException {
+		String urlRead = readURL(projectURL + "/files?page=" + page, true);
+		if(urlRead.split("<span class=\"b-pagination-item s-active active\">").length > 1 && Integer.valueOf(urlRead.split("<span class=\"b-pagination-item s-active active\">")[1].split("</span>")[0]) < page) {
+			return;
+		}
+		String[] urlReadLibs = urlRead.split("<tr class=\"project-file-list-item\">");
+		for(int i = 1; i < urlReadLibs.length; i++) {
+			if(urlReadLibs[i].split("<span class=\"version-label\">")[1].split("</span>")[0].equals(MCVersion)) {
+				calculate("https://minecraft.curseforge.com" + urlReadLibs[i].split("<a class=\"overflow-tip twitch-link\" href=\"")[1].split("\"")[0], waitMsg, ctx, list);
+				return;
+			}
+		}
+		addLatestToList(projectURL, MCVersion, waitMsg, ctx, list, page++);
+	}
+	
+	/**
+	 * Used to get the data a URL holds.
+	 * @param url The url to uses
+	 * @param simulate Should the program <u>actually</u> download the file
+	 * @return the data that the url points to. Usally a webpage
+	 * @throws CommandException 
+	 */
+	private String readURL(String url, boolean simulate) throws CommandException {
+		try {
+			InputStream urlStream = new URL(url).openStream();
+			String urlRead = "";
+			int len = urlStream.read();
+			if(simulate) {
+				while(len != -1) {
+					urlRead += (char)len;
+					len = urlStream.read();
+				}
+			}
+			return urlRead;
+		} catch (IOException e) {
+			throw new CommandException(e);
+		}
+	}
+	
+	/**
+	 * Attempt to edit the wait message, if "-i" has been set
+	 * @param waitMsg The wait message
+	 * @param newText New text to put
+	 * @param ctx the context for the message
+	 */
+	private void editWaitMessage(IMessage waitMsg, String newText, CommandContext ctx) {
+		if(!ctx.hasFlag(FLAG_INFO)) {
+			return;
+		}
+		try {
+			RequestBuffer.request(() -> waitMsg.edit(newText));
+		} catch (RateLimitException e) {
+			log.error("Unable to change " + waitMsg.getContent() + " to " + newText + e.getMessage());
+		}
+	}
+
+	@Override
+	public String getDescription() {
+		return "Displays the correct gradle infomation for a file on the curseforge maven (https://minecraft.curseforge.com/api/maven). Takes into account dependencies";
```

Comment:
Including "correct" seems unnecessary here. It implies there is a less correct version of this comment output elsewhere. Something along the lines of "Generates a list of all Gradle dependencies for the given CurseForge project."